File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        for(int i : arr) {
            for(int j = 0; j < i; j++) {
                answer.add(i);
            }
        }
        
        return answer.stream()
                     .mapToInt(Integer::intValue)
                     .toArray();
    }
}

// 다른 풀이
return Arrays.stream(arr)
             .boxed()
             .flatMap(num -> Collections.nCopies(num, num).stream())
             .collect(Collectors.toList());

File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.py
def solution(arr):
    answer = []
    
    for i in arr:
        for j in range(0, i):
            answer.append(i)
    
    return answer

# 다른 풀이
return [i for i in arr for j in range(i)]

# 다른 풀이
answer = []
for num in arr:
    answer.extend([n]*n)
return answer
